app/controllers/api/v0/highlights_controller.rb

```diff
@@ -10,12 +10,14 @@ def create
     inbound_binding, error = bind(params.require(:highlight), Api::V0::Bindings::NewHighlight)
     render(json: error, status: error.status_code) and return if error
 
-    created_highlight = service_limits.with_create_protection do |user|
-      inbound_binding.create_model!(user_id: user.id)
+    source_id = params[:highlight][:source_id]
+    Highlight.with_advisory_lock("#{current_user_uuid}#{source_id}") do
+      created_highlight = service_limits.with_create_protection do |user|
+        inbound_binding.create_model!(user_id: user.id)
+      end
+      response_binding = Api::V0::Bindings::Highlight.create_from_model(created_highlight)
+      render json: response_binding, status: :created
```

Comment:
Can we pull the response stuff out of the advisory lock so that the lock is released sooner?